tests/suites/storage/charm_storage.sh

```diff
@@ -0,0 +1,142 @@
+	echo "Block loop disk 2 PASSED"
+	# remove the application
+	juju remove-application dummy-storage-lp
+	wait_for "{}" ".applications"
+
+	echo "Assessing filesystem tmpfs"
+	juju deploy -m "${model_name}" ./testcharms/charms/dummy-storage-tp --series jammy --storage data=tmpfs,1G
+	wait_for "dummy-storage-tp" ".applications"
+	# assert the storage kind name
+	juju list-storage --format json | jq '.storage[] | .kind' | check "filesystem"
+	# assert the storage label name
+	juju list-storage --format json | jq '.storage | keys | .[] ' | check "data/3"
+	# assert the unit attachment name
+	juju list-storage --format json | jq '.storage[] | .attachments | .units | keys | .[] ' | check "dummy-storage-tp/0"
+	# assert the attached unit state
+	juju list-storage --format json | jq '.storage[] | .attachments | .units[] |.life' | check "alive"
+	echo "Filesystem tmpfs PASSED"
+	# remove the application
+	juju remove-application dummy-storage-tp
+	wait_for "{}" ".applications"
+
+	echo "Assessing filesystem"
+	juju deploy -m "${model_name}" ./testcharms/charms/dummy-storage-np --series jammy --storage data=1G
+	wait_for "dummy-storage-np" ".applications"
+	# assert the storage kind name
+	juju list-storage --format json | jq '.storage[] | .kind' | check "filesystem"
+	# assert the storage label name
+	juju list-storage --format json | jq '.storage | keys | .[]' | check "data/4"
+	# assert the unit attachment name
+	juju list-storage --format json | jq '.storage[] | .attachments | .units | keys | .[]' | check "dummy-storage-np/0"
+	# assert the attached unit state
+	juju list-storage --format json | jq '.storage[] | .attachments | .units[] | .life' | check "alive"
+	echo "Filesystem PASSED"
+	# remove application
+	juju remove-application dummy-storage-np
+	wait_for "{}" ".applications"
+	# We remove storage data/4 since in Juju 2.3+ it is persistent. Otherwise it will interfere with the next test's results
+	juju remove-storage data/4
+
+	echo "Assessing multiple filesystem, block, rootfs, loop"
+	juju deploy -m "${model_name}" ./testcharms/charms/dummy-storage-mp --series jammy --storage data=1G
+	wait_for "dummy-storage-mp" ".applications"
+	# assert the storage kind name
+	juju list-storage --format json | jq '.storage | .["data/5"] | .kind' | check "filesystem"
+	# assert the storage label name
+	juju list-storage --format json | jq '.storage | keys | .[0]' | check "data/5"
+	# assert the unit attachment name
+	juju list-storage --format json | jq '.storage | .["data/5"] | .attachments | .units | keys | .[]' | check "dummy-storage-mp/0"
+	# assert the attached unit state
+	juju list-storage --format json | jq '.storage | .["data/5"]  | .attachments | .units[] | .life' | check "alive"
+	echo "Multiple filesystem, block, rootfs, loop PASSED"
+	# remove application
+	juju remove-application dummy-storage-mp
+	wait_for "{}" ".applications"
+	echo "All charm storage tests PASSED"
+
```

Comment:
when you will add `ensure`, don't forget to `destroy_model`

Reply:
Fixed.